### PR TITLE
Add test cases for JSON resources marshaling

### PR DESCRIPTION
**Pull request by mohitDehuliya**
1. repo_hooks.go

### CODE REVIEW

File: github/repos_hooks_test.go [hunk 1]
```diff
@@ -296,3 +296,261 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
...
+		"active": true,
+		"events": ["1","2","3"],
+		"config":{
+			"thing": "@123"
+		}
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+func TestBranchHook_Marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/repos_hooks_test.go:514]: Please add blank line between tests (before this line).

File: github/repos_hooks_test.go [hunk 2]
```diff
@@ -296,3 +296,261 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
 	_, err := client.Repositories.TestHook(ctx, "%", "%", 1)
 	testURLParseError(t, err)
 }
+func TestBranchWebHookPayload_Marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/repos_hooks_test.go:299]: Please add blank line between tests (before this line).

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @mohitDehuliya!

File: github/repos_hooks_test.go (hunk 1 above)
Comment (mohitDehuliya) [reply github/repos_hooks_test.go:514]: added thanks

File: github/repos_hooks_test.go (hunk 2 above)
Comment (mohitDehuliya) [reply github/repos_hooks_test.go:299]: added thanks

File: github/repos_hooks_test.go [hunk 3]
```diff
@@ -296,3 +296,263 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
...
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchWebHookCommit_Marshal(t *testing.T) {
+	testJSONMarshal(t, &WebHookPayload{}, "{}")
```
Comment (gmlewis) [inline github/repos_hooks_test.go:451]:
```suggestion
	testJSONMarshal(t, &WebHookCommit{}, "{}")
```

File: github/repos_hooks_test.go [hunk 4]
```diff
@@ -296,3 +296,263 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
...
+		"message":  "WebHookCommit",
+		"modified": ["abc", "efg", "erd"],
+		"removed":  ["cmd", "rti", "duv"]
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchcreateHookRequest_Marshal(t *testing.T) {
+	testJSONMarshal(t, &WebHookAuthor{}, "{}")
```
Comment (gmlewis) [inline github/repos_hooks_test.go:493]:
```suggestion
	testJSONMarshal(t, &createHookRequest{}, "{}")
```

File: github/repos_hooks_test.go [hunk 5]
```diff
@@ -296,3 +296,263 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
...
+		"id":       "1",
+		"message":  "WebHookCommit",
+		"modified": ["abc", "efg", "erd"],
+		"removed":  ["cmd", "rti", "duv"]
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchcreateHookRequest_Marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/repos_hooks_test.go:492]:
```suggestion
func TestBranchCreateHookRequest_Marshal(t *testing.T) {
```

Comment (mohitDehuliya) [reply github/repos_hooks_test.go]: thanks resolved

File: github/repos_hooks_test.go (hunk 4 above)
Comment (mohitDehuliya) [reply github/repos_hooks_test.go]: thanks resolved

File: github/repos_hooks_test.go (hunk 3 above)
Comment (mohitDehuliya) [reply github/repos_hooks_test.go]: thanks resolved

**Review by gmlewis — APPROVED** [review]
Thank you, @mohitDehuliya !
LGTM.
Merging.

**Pull request merged**